### PR TITLE
Form chapter full rewrite

### PR DESCRIPTION
**Pull request by nicozanf**
Also, small related changes on DAL chapter

### CODE REVIEW

File: docs/chapter-07.rst [hunk 1]
```diff
@@ -930,18 +930,20 @@ only for fields of type “string”. ``uploadfield``, ``authorize``, and
    records are inserted in a single transaction.
 -  ``required`` tells the DAL that no insert should be allowed on this
    table if a value for this field is not explicitly specified.
--  ``requires`` is a validator or a list of validators. This is not used
-   by the DAL, but it is used by ``Form``. The default validators for
-   the given types are shown in the next section.
+-  ``requires`` is a **validator** or a list of validators. This is not used
+   by the DAL, but instead it is used by ``Form`` (this will be explained
+   better on the :ref:`Forms` chapther). The default validators for
```
Comment (agavgavi) [inline docs/chapter-07.rst:935]: Chapter is misspelled here

File: docs/chapter-12.rst [hunk 2]
```diff
@@ -2,80 +2,1457 @@
...
+
+
+-  ``Form`` is a class contained in the ``py4web.utils.form`` module
+-  it's possible to use **form validators** like ``IS_NOT_EMPTY``, see :ref:`Form validation` later.
+   They are imported from the ``pydal.validators`` module
+-  it's normally important to use both the **GET** and the **POST** methods in the action where
+   the form is contained
+
+
+This example it's not so useful because it's not using a database, a template or the session management.
```
Comment (agavgavi) [inline docs/chapter-12.rst:111]: "This example **is** not so useful" sounds better than "This example it's not so useful."

File: docs/chapter-12.rst [hunk 3]
```diff
@@ -2,80 +2,1457 @@
...
+
+
+Built-in validators have constructors that take an ``error_message`` argument:
+
+.. code:: python
+
+    IS_NOT_EMPTY(error_message='cannot be empty!')
+
+
+It'optional and it allows you to override the default error message for any validator.
```
Comment (agavgavi) [inline docs/chapter-12.rst:239]: It's optional with a space between the ' and optional.

**Review by agavgavi — COMMENTED** [review]
Just some spell checking and flow issues I saw. Also there are references to SQLFORM in some of the validators. Is that accurate?

**Comment by nicozanf** [comment]
Andrew,  many thanks! can you make a welcome fixing PR - and join us on discord? ;-)

**Comment by agavgavi** [comment]
> Andrew,  many thanks! can you make a welcome fixing PR - and join us on discord? ;-)

Sadly I'm very busy today with family all day. But maybe tomorrow I can make a pass if it isn't fixed by then.

**Pull request merged**